crates/ethereum-forks/src/hardfork/ethereum.rs

```diff
@@ -0,0 +1,445 @@
+            _ => None,
+        }
+    }
+
+    /// Retrieves the activation timestamp for the specified hardfork on the Holesky testnet.
+    pub const fn holesky_activation_timestamp(&self) -> Option<u64> {
+        #[allow(unreachable_patterns)]
+        match self {
+            Self::Shanghai => Some(1696000704),
+            Self::Cancun => Some(1707305664),
+            Self::Frontier |
+            Self::Homestead |
+            Self::Dao |
+            Self::Tangerine |
+            Self::SpuriousDragon |
+            Self::Byzantium |
+            Self::Constantinople |
+            Self::Petersburg |
+            Self::Istanbul |
+            Self::MuirGlacier |
+            Self::Berlin |
+            Self::London |
+            Self::ArrowGlacier |
+            Self::GrayGlacier |
+            Self::Paris => Some(1695902100),
+            _ => None,
+        }
+    }
+
+    /// Retrieves the activation timestamp for the specified hardfork on the Arbitrum Sepolia
+    /// testnet.
+    pub const fn arbitrum_sepolia_activation_timestamp(&self) -> Option<u64> {
+        #[allow(unreachable_patterns)]
+        match self {
+            Self::Frontier |
+            Self::Homestead |
+            Self::Dao |
+            Self::Tangerine |
+            Self::SpuriousDragon |
+            Self::Byzantium |
+            Self::Constantinople |
+            Self::Petersburg |
+            Self::Istanbul |
+            Self::MuirGlacier |
+            Self::Berlin |
+            Self::London |
+            Self::ArrowGlacier |
+            Self::GrayGlacier |
+            Self::Paris => Some(1692726996),
+            Self::Shanghai => Some(1706634000),
+            // Hardfork::ArbOS11 => Some(1706634000),
+            Self::Cancun => Some(1709229600),
+            // Hardfork::ArbOS20Atlas => Some(1709229600),
+            _ => None,
+        }
+    }
+
+    /// Retrieves the activation timestamp for the specified hardfork on the Arbitrum One mainnet.
+    pub const fn arbitrum_activation_timestamp(&self) -> Option<u64> {
+        #[allow(unreachable_patterns)]
+        match self {
+            Self::Frontier |
+            Self::Homestead |
+            Self::Dao |
+            Self::Tangerine |
+            Self::SpuriousDragon |
+            Self::Byzantium |
+            Self::Constantinople |
+            Self::Petersburg |
+            Self::Istanbul |
+            Self::MuirGlacier |
+            Self::Berlin |
+            Self::London |
+            Self::ArrowGlacier |
+            Self::GrayGlacier |
+            Self::Paris => Some(1622240000),
+            Self::Shanghai => Some(1708804873),
+            // Hardfork::ArbOS11 => Some(1708804873),
+            Self::Cancun => Some(1710424089),
+            // Hardfork::ArbOS20Atlas => Some(1710424089),
+            _ => None,
+        }
+    }
+
+    /// Ethereum mainnet list of hardforks.
+    pub fn mainnet() -> ChainHardforks {
+        ChainHardforks::new(vec![
```

Comment:
can we make this and the other chain hardforks fns an iterator over `EthereumHardfork, condition` so we don't type erase here,

and in the lazies we can map +collect